client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -123,8 +126,16 @@
         this.source = uri.getSource();
         this.extraCredentials = uri.getExtraCredentials();
         this.compressionDisabled = uri.isCompressionDisabled();
-        this.assumeLiteralNamesInMetadataCallsForNonConformingClients = uri.isAssumeLiteralNamesInMetadataCallsForNonConformingClients();
-        this.assumeLiteralUnderscoreInMetadataCallsForNonConformingClients = uri.isAssumeLiteralUnderscoreInMetadataCallsForNonConformingClients();
+
+        if (uri.isAssumeLiteralNamesInMetadataCallsForNonConformingClients()) {
+            logger.log(Level.WARNING, "Connection config assumeLiteralNamesInMetadataCallsForNonConformingClients is deprecated, please use " +
+                    "assumeLiteralUnderscoreInMetadataCallsForNonConformingClients. Enabling assumeLiteralUnderscoreInMetadataCallsForNonConformingClients.");
+            this.assumeLiteralUnderscoreInMetadataCallsForNonConformingClients = true;
+        }
+        else {
```

Comment:
why `else` being added here?

Reply:
The logic behind both params isn't _exactly_ the same but they were both introduced to solve the same problem for the user. I am making the assumption that if any clients are using the old param `assumeLiteralUnderscoreInMetadataCallsForNonConformingClients`, they should be okay to switch to the new one.   

So if they have set the old param `isAssumeLiteralNamesInMetadataCallsForNonConformingClients` to true warn them and set the new param to true for them. I have it in an `else` because if they are still using the deprecated param, we don't need to check the new param, it would likely be `false` in this case anyway. 

More details of my approach are in the description, but some are suggestion I keep the old param and its logic but just remove it from the docs. What do you think?

Reply:
One cannot set both params on (validation error), this is why the code didn't have `else` here and we shouldn't add it when deprecating one of them

Reply:
I realized I used an else because the variable is final, and Java has no way of knowing that the user can't set both configs to true, so it complains about possibly setting a final  variable twice. I have delegated to a tmp boolean instead. 

In general, do you have thoughts on this approach, or should do what @hashhar suggest and just keep the old logic, but just remove the old config from the docs.

Reply:
I added old logic back. The else block no longer exists.